[SPARK-15951] Change Executors Page to use datatables to support sorting columns and searching

kishorvpatil opened the pull request:
1. Create the executorspage-template.html for displaying application information in datables.
2. Added REST API endpoint "allexecutors" to be able to see all executors created for particular job.
3. The executorspage.js uses jQuery to access the data from /api/v1/applications/appid/allexecutors REST API, and use DataTable to display executors for the application. It also, generates summary of dead/live and total executors created during life of the application.
4. Similar changes applicable to Executors Page on history server for a given application.

Snapshots for how it looks like now:
<img width="938" alt="screen shot 2016-06-14 at 2 45 44 pm" src="https://cloud.githubusercontent.com/assets/6090397/16060092/ad1de03a-324b-11e6-8469-9eaa3f2548b5.png">

New Executors Page screenshot looks like this:
<img width="1436" alt="screen shot 2016-06-15 at 10 12 01 am" src="https://cloud.githubusercontent.com/assets/6090397/16085514/ee7004f0-32e1-11e6-9340-33d91e407f2b.png">

kishorvpatil committed:
Add AllExecutors REST Endpoint

kishorvpatil committed:
Adding ExecutorsPage template

kishorvpatil committed:
History page for Executors use mustache template

AmplabJenkins commented:
Can one of the admins verify this patch?

ajbozarth commented:
Can you post a screenshot of how it looks after the change? I'll take a look through the code when I have a chance. Also @zhuoliu want to take a look since you were the one who added JQuery DataTables to the History Server

kishorvpatil commented:
@ajbozarth , Thank you for loooking into this patch. I have added screenshot with annotation. I hope that helps.

tgravescs commented:
Jenkins, ok to test

SparkQA commented:
**[Test build #60584 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/60584/consoleFull)** for PR 13670 at commit [`56f3901`](https://github.com/apache/spark/commit/56f3901bcebdd158ba607fca20e8ef92b2468eb5).

SparkQA commented:
**[Test build #60584 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/60584/consoleFull)** for PR 13670 at commit [`56f3901`](https://github.com/apache/spark/commit/56f3901bcebdd158ba607fca20e8ef92b2468eb5).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/60584/
Test FAILed.

tgravescs commented:
Jenkins, test this please

SparkQA commented:
**[Test build #60601 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/60601/consoleFull)** for PR 13670 at commit [`56f3901`](https://github.com/apache/spark/commit/56f3901bcebdd158ba607fca20e8ef92b2468eb5).

SparkQA commented:
**[Test build #60601 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/60601/consoleFull)** for PR 13670 at commit [`56f3901`](https://github.com/apache/spark/commit/56f3901bcebdd158ba607fca20e8ef92b2468eb5).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/60601/
Test FAILed.

ajbozarth commented:
Jenkins, test this please

SparkQA commented:
**[Test build #60612 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/60612/consoleFull)** for PR 13670 at commit [`56f3901`](https://github.com/apache/spark/commit/56f3901bcebdd158ba607fca20e8ef92b2468eb5).

SparkQA commented:
**[Test build #60612 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/60612/consoleFull)** for PR 13670 at commit [`56f3901`](https://github.com/apache/spark/commit/56f3901bcebdd158ba607fca20e8ef92b2468eb5).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/60612/
Test PASSed.

zhuoliu commented:
Generally looks great to me, I am good with it as long as @tgravescs verifies. One minor concern is that we may move the utility functions from executorspage.js and historypage.ps to a common location.

ajbozarth commented:
My comment didn't post last night:
I haven't finished reading through your code yet, but after checking it out and starting up the web ui I get the following error on the executors page.
![screen shot 2016-06-15 at 6 04 26 pm](https://cloud.githubusercontent.com/assets/13952758/16127829/f6f7528c-33b2-11e6-99dd-34f7b9d5eb92.png)

ajbozarth commented:
I agree with @zhuoliu we should move common/util functions into a datatables util js.

For the above error, I pinpointed that it is baseURI that is null. I also never see the "Loading Executors Page" message i see in the code. I have also checked that the page doesn't load in any browser. @zhuoliu have you tried checking out and running the code?

kishorvpatil committed:
Merge branch 'master' of git://git.apache.org/spark into SPARK-15951

SparkQA commented:
**[Test build #61070 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/61070/consoleFull)** for PR 13670 at commit [`3b4962a`](https://github.com/apache/spark/commit/3b4962aa935aa14dbd5d04b1a894de945709319a).

SparkQA commented:
**[Test build #61070 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/61070/consoleFull)** for PR 13670 at commit [`3b4962a`](https://github.com/apache/spark/commit/3b4962aa935aa14dbd5d04b1a894de945709319a).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/61070/
Test PASSed.

kishorvpatil committed:
Remove unused javascript methods

SparkQA commented:
**[Test build #61387 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/61387/consoleFull)** for PR 13670 at commit [`c73e905`](https://github.com/apache/spark/commit/c73e905c092adfaf3faa5cb3f09d4d5a7cc071f9).

kishorvpatil commented:
@zhuoliu , i have removed unwanted, unused javascript functions in the executorspage.js. There are no duplicate functions now between historyserver.js and this script..

kishorvpatil commented:
@ajbozarth, I do not see how I could produce the exception you see. Please help me reproduce this so that I can address it.

ajbozarth commented on docs/monitoring.md:
```diff
@@ -288,7 +288,11 @@ where `[base-app-id]` is the YARN application ID.
+    <td>A list of all active executors for the given application.</td>
+  </tr>
+  <tr>
+    <td><code>/applications/[app-id]/executors</code></td>
```
These are duplicate end points: `/applications/[app-id]/executors`

tgravescs replied:
yep, I think you meant allexecutors

kishorvpatil replied:
fixed

ajbozarth commented:
I'll re-checkout later today and see if I can reproduce the error for you

ajbozarth commented:
I get the error by just starting up a spark-shell, but I only get it on Safari, it works fine on Chrome and Firefox. I'll make a line note of what I see going wrong

ajbozarth commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+
+function createRESTEndPoint() {
+    var parser = document.createElement('a');
+    var words = parser.baseURI.split('/');
```
In Safari baseURI is null, it seems to be how Safari handles document.createElement, you may want to research it and possibly find another way to get the URL, there are no other uses of baseURI in the Spark code base.

kishorvpatil replied:
As mentioned, I switched it to use `document.baseURI` which works with Safari as well.

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage-template.html:
```diff
@@ -0,0 +1,102 @@
+            <span data-toggle="tooltip" data-placement="above"
+                  title="Memory used / total available memory for storage of data like RDD partitions cached in memory.">
+              Storage Memory</span></th>
+          <th><span data-toggle="tooltip" data-placement="above" title="Disk Used">Disk Used</span></th>
```
these tool tips (with placement above) are showing up in the upper left corner of the table for me rather then right above the column

kishorvpatil replied:
Fixed

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage-template.html:
```diff
@@ -0,0 +1,102 @@
+          <th>
+            <span data-toggle="tooltip" data-placement="right" title="ID of the executor">Executor ID</span></th>
+          <th>
+            <span data-toggle="tooltip" data-placement="above" title="Address">Address</span></th>
```
we probably don't need a tool tip if its just re-iterating the column name.  We didn't have them before so unless we are going to put more useful description I would say just remove.

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+ */
+
+// this function works exactly the same as UIUtils.formatDuration
+function formatDuration(milliseconds, type) {
```
this is the pretty much the same with historypage.js could we easily make a utils.js and include in both.

I also don't really see the type being used here, it seems everywhere its called it already has check for type === display

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+    }
+}
+
+function formatBytes(bytes,type) {
```
similar here, type isn't really used.

tgravescs replied:
sorry I'm assuming I'm wrong here because we have things like:
+                        {data: 'totalInputBytes', render: formatBytes },

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+    var parser = document.createElement('a');
+    var words = parser.baseURI.split('/');
+    var ind = words.indexOf("proxy");
+    if(ind > 0) {
```
nit, space between if and (

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+    var parser = document.createElement('a');
+    var words = parser.baseURI.split('/');
+    var ind = words.indexOf("proxy");
+    if(ind > 0) {
```
nit space after if, note this is throughout file

tgravescs commented on core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala:
```diff
@@ -54,78 +53,16 @@ private[ui] class ExecutorsPage(
+    UIUtils.headerSparkPage("Executors", content, parent, useDataTables = true)
   }
 
   /** Render an HTML row representing an executor */
```
it seems like there is a bunch of other things in here that were used by render that can be removed. execRow, execSummaryRow, taskData, etc.

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+    return  activeTasks > 0 ? ("hsla(240, 100%, 50%, " + activeTasksAlpha(activeTasks, maxTasks) + ")") : "";
+}
+
+function activeTasksColor(activeTasks, maxTasks) {
```
not used, is this supposed to be color:white?

tgravescs replied:
actually looks like its hardcoded below with css

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+        ("hsla(0, 100%, 50%, " + failedTasksAlpha(failedTasks, totalTasks) + ")") : "";
+}
+
+function failedTasksColor(failedTasks, totalTasks){
```
not used

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+                        {data: 'hostPort'},
+                        {data: 'isActive', render: formatStatus },
+                        {data: 'rddBlocks'},
+                        {data: function(row, type) { return type === 'display' ? (formatBytes(row.memoryUsed, type) + ' / ' + formatBytes(row.maxMemory, type)) : row.memoryUsed; } },
```
this is showing up with 3 digits after .
39.266 KB / 384.093 MB

previously was using Utils.bytesToString which made it 1 digit after . which I think is more readable so can we change to that

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+    });
+
+    executorsSummary = $("#active-executors");
+    searchString = executorsSummary["context"]["location"]["search"];
```
I'm not seeing this being used?

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+      var deadTotalShuffleRead = 0;
+      var deadTotalShuffleWrite = 0;
+
+        response.forEach(function (exec) {
```
indentation is off

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+          allCompletedTasks, "allTotalTasks" : allTotalTasks, "allTotalDuration" : allTotalDuration,
+          "allTotalGCTime" : allTotalGCTime, "allTotalInputBytes" : allTotalInputBytes,
+          "allTotalShuffleRead" : allTotalShuffleRead, "allTotalShuffleWrite" : allTotalShuffleWrite };
+      var activeSummary = { "execCnt" : ( "Active(" + activeExecCnt + ")"), "allRDDBlocks" : activeRDDBlocks, "allMemoryUsed" : activeMemoryUsed,
```
The labels here don't match... ie "allRDDBlocks" : activeRDDBlocks, copy and past error from totalSummary

tgravescs replied:
ok ignore my comments, that is column name

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+          activeCompletedTasks, "allTotalTasks" : activeTotalTasks, "allTotalDuration" : activeTotalDuration,
+            "allTotalGCTime" : activeTotalGCTime, "allTotalInputBytes" : activeTotalInputBytes,
+            "allTotalShuffleRead" : activeTotalShuffleRead, "allTotalShuffleWrite" : activeTotalShuffleWrite };
+      var deadSummary = { "execCnt" : ( "Dead(" + deadExecCnt + ")" ), "allRDDBlocks" : deadRDDBlocks, "allMemoryUsed" : deadMemoryUsed,
```
The labels here don't match... ie "allRDDBlocks" : deadRDDBlocks, copy and past error from totalSummary

tgravescs replied:
ok ignore my comments, that is column name

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage-template.html:
```diff
@@ -0,0 +1,102 @@
+-->
+
+<script id="executors-summary-template" type="text/html">
+  <h3 style="clear: left; display: inline-block;">Summary</h3>
```
can we make the size of Summary slightly smaller because right now its the same size as "Executors" on the top of the page.

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage-template.html:
```diff
@@ -0,0 +1,102 @@
+      </table>
+    </div>
+  </div>
+  <h3 style="clear: left; display: inline-block;">Executors</h3>
```
can we make the size of Summary slightly smaller because right now its the same size as "Executors" on the top of the page.

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,410 @@
+        var conf = {
+                    "data": response,
+                    "columns": [
+                        //{data: 'id'},
```
remove

tgravescs commented on core/src/main/scala/org/apache/spark/status/api/v1/AllExecutorListResource.scala:
```diff
@@ -0,0 +1,43 @@
+*/
+package org.apache.spark.status.api.v1
+
+import javax.ws.rs.{GET, PathParam, Produces}
```
PathParam doesn't appear to be used

tgravescs commented:
Made a pass through and some mostly minor comments.

kishorvpatil committed:
Fix the tooltip placement for detailed executors table headers

SparkQA commented:
**[Test build #62191 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62191/consoleFull)** for PR 13670 at commit [`2453e21`](https://github.com/apache/spark/commit/2453e21d9ef415725c13855555876abce4b20344).

kishorvpatil committed:
Fix the baseURI reference for safari

kishorvpatil commented:
@ajbozarth, I have fixed the issue with safari exception. Its because `parser.baseURI` is `null`. I switched to `document.baseURI`. It looks fine on testing. Let me know if you still see this issue.

SparkQA commented:
**[Test build #62194 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62194/consoleFull)** for PR 13670 at commit [`55419aa`](https://github.com/apache/spark/commit/55419aad1e4db20cbed136c97a51d11a97fff320).

ajbozarth commented:
@kishorvpatil thanks, I'll check it out

kishorvpatil committed:
Removed unused imports and refactor to use in-line references

kishorvpatil committed:
Lower heading level for summary to 4

SparkQA commented:
**[Test build #62195 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62195/consoleFull)** for PR 13670 at commit [`3454193`](https://github.com/apache/spark/commit/345419398cf82c1e9b1360a622b9b874154677e7).

SparkQA commented:
**[Test build #62191 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62191/consoleFull)** for PR 13670 at commit [`2453e21`](https://github.com/apache/spark/commit/2453e21d9ef415725c13855555876abce4b20344).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62191/
Test PASSed.

SparkQA commented:
**[Test build #62194 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62194/consoleFull)** for PR 13670 at commit [`55419aa`](https://github.com/apache/spark/commit/55419aad1e4db20cbed136c97a51d11a97fff320).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62194/
Test PASSed.

SparkQA commented:
**[Test build #62195 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62195/consoleFull)** for PR 13670 at commit [`3454193`](https://github.com/apache/spark/commit/345419398cf82c1e9b1360a622b9b874154677e7).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62195/
Test PASSed.

kishorvpatil committed:
Fixed decimal precision for disk usage displayed

SparkQA commented:
**[Test build #62258 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62258/consoleFull)** for PR 13670 at commit [`5cf514d`](https://github.com/apache/spark/commit/5cf514d4a9047955028b450d33263cb863987112).

kishorvpatil committed:
Refactor common method out and reformat script files

kishorvpatil committed:
Refactor to use common methods from utils.js

SparkQA commented:
**[Test build #62263 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62263/consoleFull)** for PR 13670 at commit [`eb96c5d`](https://github.com/apache/spark/commit/eb96c5d89077041418c3c667e579373a9e1ddedb).

SparkQA commented:
**[Test build #62263 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62263/consoleFull)** for PR 13670 at commit [`eb96c5d`](https://github.com/apache/spark/commit/eb96c5d89077041418c3c667e579373a9e1ddedb).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62263/
Test FAILed.

kishorvpatil committed:
Remove unused method

kishorvpatil committed:
Remove unused variables

SparkQA commented:
**[Test build #62267 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62267/consoleFull)** for PR 13670 at commit [`fe12d59`](https://github.com/apache/spark/commit/fe12d599dc95dd7346f0f2f3ee04cdce039a05e7).

SparkQA commented:
**[Test build #62267 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62267/consoleFull)** for PR 13670 at commit [`fe12d59`](https://github.com/apache/spark/commit/fe12d599dc95dd7346f0f2f3ee04cdce039a05e7).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62267/
Test FAILed.

kishorvpatil committed:
Fix REST API documenation for all executors

SparkQA commented:
**[Test build #62269 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62269/consoleFull)** for PR 13670 at commit [`286d6d8`](https://github.com/apache/spark/commit/286d6d8b40e5e5840837e57c9e4a9aefaca19691).

SparkQA commented:
**[Test build #62269 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62269/consoleFull)** for PR 13670 at commit [`286d6d8`](https://github.com/apache/spark/commit/286d6d8b40e5e5840837e57c9e4a9aefaca19691).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62269/
Test FAILed.

SparkQA commented:
**[Test build #62258 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62258/consoleFull)** for PR 13670 at commit [`5cf514d`](https://github.com/apache/spark/commit/5cf514d4a9047955028b450d33263cb863987112).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62258/
Test PASSed.

kishorvpatil committed:
Remove unwanted variables/methods from javascript

SparkQA commented:
**[Test build #62274 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62274/consoleFull)** for PR 13670 at commit [`faf814f`](https://github.com/apache/spark/commit/faf814fe5d710ecbc77d10cbf9028ca5f0aa5ebb).

kishorvpatil commented:
@tgravescs, I think addressed the comments mentioned above.

SparkQA commented:
**[Test build #62274 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62274/consoleFull)** for PR 13670 at commit [`faf814f`](https://github.com/apache/spark/commit/faf814fe5d710ecbc77d10cbf9028ca5f0aa5ebb).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62274/
Test PASSed.

ajbozarth commented:
Thanks for all the fixes, I checked it out and am in the middle of looking at it. It looks great on the history server for me but the page just doesn't load or show and form of errors when i try to load it in the Web UI though. I'll continue to try to figure out why, but it's consistent across all browsers. I did notice your two screen shots were of the history server and through a proxy, have you tried running your code on the standard web ui?

ajbozarth commented:
Follow up to my previous post, this works on yarn and the history server but not on standalone, as a user I can't figure out what's broken.

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,458 @@
+function createTemplateURI() {
+    var parser = document.createElement('a');
+    var words = document.baseURI.split('/');
+    var ind = words.indexOf("proxy");
```
I'm guessing this is the issue with it not working in standalone mode.  It wouldn't have proxy or history in it necessarily.  Even on yarn it doesn't necessarily have the proxy in it (ie its configurable on yarn whether to use it or not).

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,458 @@
+    }
+}
+
+function formatBytes(bytes, type) {
```
this feels like something we should put in utils.  history isn't using it now but others would need to once they are ported.

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,458 @@
+            var conf = {
+                "data": response,
+                "columns": [
+                    //{data: 'id'},
```
remove commented out

tgravescs commented on docs/monitoring.md:
```diff
@@ -288,7 +288,11 @@ where `[base-app-id]` is the YARN application ID.
+  </tr>
+  <tr>
+    <td><code>/applications/[app-id]/allexecutors</code></td>
+    <td>A list of all(active/dead) executors for the given application.</td>
```
maybe change active/dead to be active and dead.

kishorvpatil committed:
Fix ExecutorsPage for standalone mode

kishorvpatil committed:
Minor documentation fix

kishorvpatil commented:
@ajbozarth , I fixed the code for standalone mode as well. Sorry for not being fully aware of possible UI invocations. I will keep this in mind for future patches.

kishorvpatil commented:
@tgravescs, fixed documentation.

SparkQA commented:
**[Test build #62343 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62343/consoleFull)** for PR 13670 at commit [`bc132b1`](https://github.com/apache/spark/commit/bc132b1c23982062fbe3c411a746a9c0eb10b380).

kishorvpatil committed:
Move formateBytes function to utils.js

SparkQA commented:
**[Test build #62346 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62346/consoleFull)** for PR 13670 at commit [`66df19f`](https://github.com/apache/spark/commit/66df19fbf564f9541abebac4ecb0d00379bcb7ae).

kishorvpatil commented:
@tgravescs , moved `formatBytes` to `utils.js`.

SparkQA commented:
**[Test build #62343 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62343/consoleFull)** for PR 13670 at commit [`bc132b1`](https://github.com/apache/spark/commit/bc132b1c23982062fbe3c411a746a9c0eb10b380).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62343/
Test PASSed.

SparkQA commented:
**[Test build #62346 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62346/consoleFull)** for PR 13670 at commit [`66df19f`](https://github.com/apache/spark/commit/66df19fbf564f9541abebac4ecb0d00379bcb7ae).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62346/
Test PASSed.

tgravescs commented:
@ajbozarth  were you able to try the latest version?

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,480 @@
+    });
+
+    executorsSummary = $("#active-executors");
+    searchString = executorsSummary["context"]["location"]["search"];
```
unused, remove

tgravescs commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,480 @@
+                var conf = {
+                    "data": response,
+                    "columns": [
+                        //{data: 'id'},
```
remove commented out line

ajbozarth commented:
Been busy the last couple days, I'll take a look at this Monday

kishorvpatil committed:
Remove commented unused js lines

kishorvpatil commented:
@tgravescs, I removed unwanted code/comment from the javascript.

SparkQA commented:
**[Test build #62472 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62472/consoleFull)** for PR 13670 at commit [`e81d45b`](https://github.com/apache/spark/commit/e81d45bb7ba530ac37398e3e78cda8054123e10f).

SparkQA commented:
**[Test build #62472 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62472/consoleFull)** for PR 13670 at commit [`e81d45b`](https://github.com/apache/spark/commit/e81d45bb7ba530ac37398e3e78cda8054123e10f).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62472/
Test PASSed.

ajbozarth commented:
Sorry I couldn't get to this today, I'll do my best to take a look in the morning

ajbozarth commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,478 @@
+    return activeTasks > 0 ? ("hsla(240, 100%, 50%, " + activeTasksAlpha(activeTasks, maxTasks) + ")") : "";
+}
+
+function activeTasksColor(activeTasks, maxTasks) {
```
This is never used

ajbozarth commented on core/src/main/resources/org/apache/spark/ui/static/executorspage.js:
```diff
@@ -0,0 +1,478 @@
+        ("hsla(0, 100%, 50%, " + failedTasksAlpha(failedTasks, totalTasks) + ")") : "";
+}
+
+function failedTasksColor(failedTasks, totalTasks) {
```
This is also never used

ajbozarth commented:
I checked out out the latest code and ran it and it looks good, I also looked through the code and outside of the two unused functions I found the code looks good too.

Im comfortable giving this a LGTM

We may want to start setting a standard way to convert pages to use DataTables, this seemed more a difficult process than I thought it would be

kishorvpatil committed:
Remove unused javascript functions

kishorvpatil commented:
@ajbozarth Thank you for review. I removed unused methods from the the javascript. 

I agree converting pages to `DataTables` is tricky and listing some standard way might be a huge help.

SparkQA commented:
**[Test build #62607 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62607/consoleFull)** for PR 13670 at commit [`87301c1`](https://github.com/apache/spark/commit/87301c1e0de1ade000fb7d05685b61ca87fc1491).

SparkQA commented:
**[Test build #62607 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/62607/consoleFull)** for PR 13670 at commit [`87301c1`](https://github.com/apache/spark/commit/87301c1e0de1ade000fb7d05685b61ca87fc1491).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/62607/
Test PASSed.

tgravescs commented:
+1, thanks @kishorvpatil @ajbozarth

nblintao commented:
Hi, @kishorvpatil. Cool improvement! I'm developing the Executors Page at #14204 based on your changes. But I have a small question about your new implementation of the tables.
In the implementation of `ExecutorsPage.scala` before, `logsExist` is checked and there will be no Logs column if it's empty for all rows. But now, I haven't found similar implementation in the `executorspage.js`. I'm not sure whether it is handled by JQuery DataTables  automatically, or I'm missing something, or this is not available for now? Thanks.

kishorvpatil commented:
@nblintao , I am not sure why we had the check of existence of logs - specially when the reference to logs is static link for `stdout` and `stderr`.
Unless there is scenario where there could be multiple log entries and dynamic list of logs- i suggest that would require coding of different solution - mere check for existence and dropping out column altogether would not help.

ajbozarth commented:
@kishorvpatil I can't believe I missed that. The logs only exist if you have logging turned on in your confs, if it's off there won't be any logs and no reason for the column. @nblintao if you want to file a JIRA either you @kishorvpatil or I can quickly fix that

tgravescs commented:
personally I don't see this as a big deal.  As long as it doesn't show the links, the column being there doesn't hurt anything and it keeps things more consistent.  Perhaps change it to say Not Available or something?   

If you thinkts its an issue lets file a separate jira for this and discuss further there.

ajbozarth commented:
I opened SPARK-16673 for the issue since it also apply to the Thread Dump column which the History Server shouldn't display.

merlintang commented:
@kishorvpatil 
you provided the function allexecutors, which is used to return the dead and active executor information. 

For the document
http://spark.apache.org/docs/latest/monitoring.html
for 
/applications/[app-id]/executors	A list of all executors for the given application.

We had better document more clearly what is meaning of functions for the 2.1 version. 
/applications/[app-id]/executors xxx
/applications/[app-id]/allexecutors  xxxx

This confused people, because our test already run into this issue.
